### PR TITLE
Bug 4820/v2

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4820

### CODE REVIEW

File: src/detect-xbits.c [hunk 1]
```diff
@@ -538,6 +542,166 @@ static int XBitsTestSig02(void)
...
+    de_ctx->flags |= DE_QUIET;
+
+    Signature *s = DetectEngineAppendSig(de_ctx,
+            "alert http any any -> any any (msg:\"ET EXPLOIT Possible Pulse Secure VPN RCE "
+            "Chain Stage 1 Inbound - Request Config Backup (CVE-2020-8260)\"; "
+            "flow:established,to_server; http.method; content:\"GET\"; http.uri; "
+            "content:\"/dana-admin/cached/config/config.cgi?type=system\"; fast_pattern; "
+            "xbits:set,ET.2020_8260.1,track ip_src,expire 10,noalert; "
+            "reference:url,packetstormsecurity.com/files/160619/"
+            "Pulse-Secure-VPN-Remote-Code-Execution.html; reference:cve,2020-8260; "
```
Comment (victorjulien) [inline src/detect-xbits.c:600]: is there a point to adding the full rule incl references and such? Seems to overdo things a bit to me.

File: src/detect-xbits.c [hunk 2]
```diff
@@ -538,6 +542,166 @@ static int XBitsTestSig02(void)
...
+    PASS;
+}
+
+static int DetectXBitsTestGoodRules(void)
+{
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+
+    const char *sigs[] = {
+        "alert http any any -> any any (content:\"abc\"; xbits:set,bit1,track ip_src;sid:1;)",
```
Comment (victorjulien) [inline src/detect-xbits.c:681]: good rules should also test spaces around the comma's

File: src/detect-xbits.c (hunk 1 above)
Comment (inashivb) [reply src/detect-xbits.c:600]: OK. I copied and pasted the entire rule that was failing.

**Comment by inashivb** [comment]
Replaced w #6889

**Pull request closed without merging**